### PR TITLE
adding ToastModule with tests

### LINKED ISSUE #46: Windows Toast Notification Support

### PR DESCRIPTION
**Pull request by cicorias**
closes #46

### CODE REVIEW

File: ReactWindows/ReactNative.Tests/Modules/Toast/ToastNotificationTests.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ReactNative.Tests.Modules.Toast
+{
+    [TestClass]
+    public class ToasNotificationTests
```
Comment (timfpark) [inline ReactWindows/ReactNative.Tests/Modules/Toast/ToastNotificationTests.cs]: Typo in name of Test class

File: ReactWindows/ReactNative.Tests/Modules/Toast/ToastNotificationTests.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            var context = new ReactApplicationContext();
+            var module = new ToastModule(context);
+
+            module.show("LONG TOAST container", 1);
+        }
+
+        [TestMethod]
+        [TestCategory("Modules")]
+        public void Send_Toas_Invalid_Duration()
```
Comment (timfpark) [inline ReactWindows/ReactNative.Tests/Modules/Toast/ToastNotificationTests.cs]: Typo in Test name

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+using System.Collections.Generic;
+using Windows.Data.Xml.Dom;
+using Windows.UI.Notifications;
+
+namespace ReactNative.Modules.Toast
+{
+    using System.Diagnostics;
+    using System.IO;
+    using System.Threading.Tasks;
+    using Map = Dictionary<string, object>;
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: Some usings on the inside, some on the outside?

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            get
+            {
+                return "ToastModule";
+            }
+        }
+
+        public Map getConstants()
+        {
+            return
+                    new Map
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: Looks like some extra whitespace.

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+        public override string Name
+        {
+            get
+            {
+                return "ToastModule";
+            }
+        }
+
+        public Map getConstants()
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: This should be overridden from the constants property.

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    using System.Threading.Tasks;
+    using Map = Dictionary<string, object>;
+    public partial class ToastModule : ReactContextNativeModuleBase
+    {
+        public ToastModule(ReactApplicationContext reactContext)
+            : base(reactContext)
+        {
+        }
+
+        private static readonly string DURATION_SHORT_KEY = "SHORT";
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: These could be const.  Also, usually consts / fields are declared before constructor.

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        [ReactMethod]
+        public void show(string message, int duration)
+        {
+            /// TODO: UWP doesn't support a duration option.
+            ToastHelper.SendToast(message, "short");
+        }
+
+    }
+
+    enum durations : int { @short, @long }
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: C# class / enum names should start with capital letter.

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+                var doc = new XmlDocument();
+                doc.LoadXml(xml);
+                return doc;
+            }
+
+        }
+
+        static async Task BuildNotificationXml(Stream stream, string title, string content, durations duration=durations.@short)
+        {
+            System.Xml.XmlWriterSettings settings = new System.Xml.XmlWriterSettings();
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: Any reason not to do "using System.Xml;"?

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+                await BuildNotificationXml(stream, title, content, duration);
+                stream.Position = 0;
+                var xml = await reader.ReadToEndAsync();
+                Debug.WriteLine("Toast XML");
+                Debug.WriteLine(xml);
+                var doc = new XmlDocument();
+                doc.LoadXml(xml);
+                return doc;
+            }
+
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: Delete extra newline.

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+                await writer.WriteStartElementAsync(null, "text", null);
+                await writer.WriteAttributeStringAsync(null, "id", null, "2");
+                await writer.WriteStringAsync(content);
+                await writer.WriteEndElementAsync();
+
+                await writer.WriteEndElementAsync();
+                await writer.FlushAsync();
+            }
+        }
+
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: Extra newline

File: ReactWindows/ReactNative.Tests/Modules/Toast/ToastNotificationTests.cs [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ReactNative.Tests.Modules.Toast
+{
+    [TestClass]
+    public class ToasNotificationTests
+    {
+
+        //[TestMethod]
```
Comment (rozele) [inline ReactWindows/ReactNative.Tests/Modules/Toast/ToastNotificationTests.cs]: Clean up commented out code?

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs (hunk 8 above)
Comment (cicorias) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: [Windows.Data.Xml.Dom.XmlDocument\(https://msdn.microsoft.com/library/windows/apps/br206173) in use in UWP vs. System.Xml.Windows.Data.Xml.Dom

File: ReactWindows/ReactNative.Tests/Modules/Toast/ToastNotificationTests.cs [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            var context = new ReactApplicationContext();
+            var module = new ToastModule(context);
+
+            module.show("LONG TOAST container", 1);
+        }
+
+        [TestMethod]
+        [TestCategory(TEST_CATEGORY)]
+        public void Send_Toas_Invalid_Duration()
```
Comment (rozele) [inline ReactWindows/ReactNative.Tests/Modules/Toast/ToastNotificationTests.cs]: Send_Toast_Invalid_Duration

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Windows.Data.Xml.Dom;
+using Windows.UI.Notifications;
+
+namespace ReactNative.Modules.Toast
+{
+    public partial class ToastModule : ReactContextNativeModuleBase
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: sealed?

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+    {
+        private static readonly string DURATION_SHORT_KEY = "SHORT";
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: const

File: ReactWindows/ReactNative/Modules/Toast/ToastModule.cs [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+        }
+    }
+
+    enum Durations : int { @short, @long }
+
+    internal class ToastHelper
+    {
+        internal static void SendToast(string message, string duration = "short")
+        {
+            var doc = MakeDoc("foobar", message).Result;
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Toast/ToastModule.cs]: "foobar"?

**Pull request merged**